pkg/roachpb/errors.go

```diff
@@ -49,6 +49,15 @@ func (e *RetryableTxnError) Error() string {
 
 var _ error = &RetryableTxnError{}
 
+// NewLocalRetryableError creates a shim RetryableTxnError that
+// reports the given cause when converted to String(). This can be
+// used to fake/force a retry at the SQL layer.
+func NewLocalRetryableError(cause string) *RetryableTxnError {
```

Comment:
Would you be ok when with exporting the `message` attribute? It's currently private and thus not visible to the `sql` package.